tests/test_decorators.py

```diff
@@ -668,3 +668,41 @@ def make_item(foo, bar):
     assert '_schema' in errors
     assert len(errors['_schema']) == 1
     assert errors['_schema'][0] == 'preloadmsg1'
+
+@pytest.mark.parametrize(
+    'decorator',
+    [
+        pre_load,
+        post_load,
+    ]
+)
+def test_decorator_strict_error_handling_with_load(decorator):
+    class ExampleSchema(Schema):
+        @decorator
+        def raise_value_error(self, item):
+            raise ValidationError({'foo': 'error'})
+
+    schema = ExampleSchema(strict=True)
+    with pytest.raises(ValidationError) as exc:
+        schema.load({})
+    assert exc.value.messages == {'foo': 'error'}
+    schema.dump(object())
+
+@pytest.mark.parametrize(
+    'decorator',
+    [
+        pre_dump,
+        post_dump,
+    ]
+)
+def test_decorator_strict_error_handling_with_dump(decorator):
+    class ExampleSchema(Schema):
+        @decorator
+        def raise_value_error(self, item):
+            raise ValidationError({'foo': 'error'})
+
+    schema = ExampleSchema(strict=True)
```

Comment:
> Did you check that it was broken before?

Yes

> Theses tests look like they would work without any modifications.

They __should__ work. So, there is a bug. See marshmallow-code/marshmallow#521

> With strict=True it throws exception from pre/post_* callbacks (that's how it worked before);

These tests check exactly that.

> (more importantly) Without strict=False it does not raise exceptions BUT returns those errors as errors field in Marshmal/Unmarshal result.

There are already tests for that https://github.com/marshmallow-code/marshmallow/blob/dev/tests/test_decorators.py#L534-L670